core/hash_map.h

```diff
@@ -261,147 +229,97 @@ class HashMap {
 	}
 
 	Element *set(const Pair &p_pair) {
-
-		Element *e = NULL;
-		if (!hash_table)
-			make_hash_table(); // if no table, make one
-		else
-			e = const_cast<Element *>(get_element(p_pair.key));
-
-		/* if we made it up to here, the pair doesn't exist, create and assign */
-
-		if (!e) {
-
-			e = create_element(p_pair.key);
-			if (!e)
-				return NULL;
-			check_hash_table(); // perform mantenience routine
-		}
-
+		Element *e = _find_or_create_element(p_pair.key);
 		e->pair.data = p_pair.data;
 		return e;
 	}
 
 	bool has(const TKey &p_key) const {
-
 		return getptr(p_key) != NULL;
 	}
 
-	/**
-	 * Get a key from data, return a const reference.
-	 * WARNING: this doesn't check errors, use either getptr and check NULL, or check
-	 * first with has(key)
-	 */
-
+	// Get data using a key
+	// WARNING: this doesn't check errors,
+	// use either getptr directly and check NULL, or check first with has(key)
```

Comment:
Why not leave multi-line comments as multi-line `/* */`? If desired, we could reduce the number of lines by moving the starting and ending ones to the same lines as the text.

Reply:
I'm just used to using single line comments, hence the change. But by trying to mainly use single line comments, it gets me to try and keep comments simple and placed where necessary.

This is a preference of comment syntax. I've changed it to single line comments in this pr but can change it back to multi-line comments, if they are preferred. As long as the syntax remains consistent within the file.

In this case, the first line could be removed entirely, as it just repeats what the function definition already says, so I'll change that.

Reply:
Agree completely on that last part. When I look at comments I don't want a restatement of the function name (unless the sentence adds more information as well). Caveats and when-to-use/when-not-to-use are great.